### DISCUSSION TITLE
Column_property with self-referential fields.

### DISCUSSION THREAD

**Discussion by nikitabray** [sqlalchemy/sqlalchemy#7210] [Usage Questions] [answered]
Hi. I already asked this on gitter but didn't get enough explanation.

I have model

```
class Node(BaseModel):
    __tablename__ = "nodes"
    
    id = Column(String, primary_key=True, unique=True, index=True)
    parent_id = Column(String, ForeignKey("nodes.id"), nullable=True)
    children = relationship("Node", lazy="immediate")
    has_children = ?
```

And i need has_children to be a column_property that get information about node's children existence. The problem is: i can't do something like `has_children = column_property(exists().where(Node.parent_id == id))` because i can't get `Node` class object from it's attributes. I also can't do `Node.has_children = column_property(exists().where(Node.parent_id == <?>))` after `Node` model was defined because i don't know what should i pass to this `<?>` placeholder. 
Somehow using children field is not possible because sometimes i don't load it from DB.

**Comment by CaselIT** [comment]
Hi,

it seems the initial message is missing some parts

**Reply by nikitabray** [reply]
Yeah, i accidentally hit submit button

**Comment by CaselIT** [comment, marked as answer]
You probably need to define has_children after the class, so that you can use an alias of Node to select from and the original node to correlate towards.

So something like this:
```py
NA = orm.aliased(Node)
Node.has_children = column_property(select(exists().select_from(NA).where(NA.parent_id == Node.id).correlate(Node)).scalar_subquery())
```

**Reply by nikitabray** [reply]
Works perfectly, thank you!
